### PR TITLE
Add tests to `ranker::Context`

### PR DESCRIPTION
**Pull request by x3ro**

### CODE REVIEW

File: xayn-ai/src/ranker/context.rs [hunk 1]
```diff
@@ -19,13 +21,82 @@ impl Context {
         Self { pos_avg, neg_max }
     }
 
-    /// Calculates context value from given LTR score, positive distance, negative distance and similarity.
-    pub(crate) fn calculate(&self, pos: f32, neg: f32) -> f32 {
+    /// Calculates context value from given positive distance, negative distance.
```
Comment (Robert-Steiner) [inline xayn-ai/src/ranker/context.rs:24]:
```suggestion
    /// Calculates score from given positive distance and negative distance.
```

File: xayn-ai/src/ranker/context.rs [hunk 2]
```diff
@@ -1,5 +1,7 @@
 use crate::{data::document_data::CoiComponent, ranker::util::Id};
 
+/// Ranking context calculation according to
+/// [the spec](https://xainag.atlassian.net/wiki/spaces/M2D/pages/770670607/Production+AI+Workflow#3.2-Context-calculations.)
```
Comment (Robert-Steiner) [inline xayn-ai/src/ranker/context.rs:4]:
```suggestion
/// The context used to calculate a document's score.
```
I like the idea of ​​linking the document, but it describes the old context calculation and that could be a bit confusing.

File: xayn-ai/src/ranker/context.rs [hunk 3]
```diff
@@ -19,13 +21,82 @@ impl Context {
...
+
+        let cxt = calc.calculate_score(8., 7.);
+        assert_approx_eq!(f32, cxt, ((1. / 3.) + (1. / 2.)) / 2.);
+    }
+
+    #[test]
+    fn test_from_cois() {
+        let cois = vec![
+            (
+                Id::from_rand(),
```
Comment (Robert-Steiner) [inline xayn-ai/src/ranker/context.rs:73]: we can also use `Id::from_u128(0)` because for the context calculation it doesn't matter if the id is unique

**Review by Robert-Steiner — COMMENTED** [review]
Looks good 👍 only found some minor things

**Comment by x3ro** [comment]
Re: the documentation link: Personally I'd probably keep it around, since it'd be easier to understand the code with outdated info than it was with no info at all. I don't feel strongly about it though.

**Review by Robert-Steiner — APPROVED** [review]
> Re: the documentation link: Personally I'd probably keep it around, since it'd be easier to understand the code with outdated info than it was with no info at all. I don't feel strongly about it though.

we could add:
```
/// https://xainag.atlassian.net/wiki/spaces/M2D/pages/770670607/Production+AI+Workflow#3.2-Context-calculations.
/// outlines the calculation of positive and negative distance factor.
```

**Comment by x3ro** [comment]
Would you put this on `Context` or `calculate_score`?

**Pull request merged**